### ISSUE TITLE
Derivative reference volumes for 4D series

### ISSUE THREAD

**Issue by effigies** [bids-standard/bids-specification#1532]
### Your idea

In fMRIPrep, we generate `_boldref.nii.gz` files from some combination of the source bold file, an associated sbref (if available) and within-space transforms (such as SDC warp). This is used for performing SDC, coregistration, visualization, and as a resampling target for a full BOLD series.

I was initially going to put this into #519 as `_boldref` and `_cbvref`, but is it more general? Does this make sense to put into [Imaging data types](https://bids-specification.readthedocs.io/en/latest/derivatives/imaging.html)?

Would something like the following ever be generated?

* `_dwiref` (cc @oesteban, @arokem, @Lestropie, @bids-standard/derivatives-mri-dwi)
* `_aslref` (cc @patsycle, @HenkMutsaerts)
* `_petref` (cc @mnoergaard @melanieganz, @bids-standard/derivatives-pet)

**Comment by tsalo** [comment]
ASLPrep generates aslref files.

**Comment by oesteban** [comment]
In the dMRI world, I think such a "dwi reference" is typically a $b = 0$ (`b0` or `bzero`). So `_dwiref` would force people to change their standard. On the other hand, I see great usefulness in standardizing `_<modality_suffix>ref` as the suffix across modalities (and I would bet a similar concept of "individual alignment reference" may exist even for EEG, MEG or fNIRS)

**Comment by effigies** [comment]
I think in cases where there is a definite physical meaning, it makes sense to have a specialized suffix. We could describe reference volumes and then simply have a table that maps the raw suffix onto the derived reference suffix:

| Data type | Modality | Derived reference |
|--|--|--|
| `func` | `bold` | `boldref` |
| `func` | `cbv` | `cbvref`|
| `dwi` | `dwi` | `b0ref` |
| `perf` | `asl` | `aslref` |
| ...

DWI stands out a little, but the concepts are still linked in the spec and will make sense to each modality's practitioners.

**Comment by oesteban** [comment]
Actually, there are occasions where the reference is calculated with
nonzero (but low) $b$ values. I'm not sure there's a big difference between
`_b0ref` and `_dwiref` while the latter is always theoretically correct and
more consistent with the other.

WDYT @yasseraleman ?

**Comment by mattcieslak** [comment]
I wonder if it would make sense someday to differentiate between `b0` and `b=0` in bids somehow. Since there are already metadata fields that include `b0` in their name and are referring to something totally different. fwiw qsiprep has been writing out _dwiref files, which are the average b=0 images

**Comment by oesteban** [comment]
If qsiprep already writes `_dwiref`, I guess that settles it right?

**Comment by effigies** [comment]
Yeah, sounds reasonable. I'll write up a PR, as it should be pretty quick.

**Comment by effigies** [comment]
Just a note that this was not an uncontroversial proposal. Additional input on #1533 would be appreciated.

**Comment by mnoergaard** [comment]
https://github.com/bids-standard/bids-specification/issues/1532#issue-1779365385 - definitely! We will also have a `_petref` to indicate the file to be used with registration between PET and anatomical. However, there may be multiple refs present, as one might be using one ref for motion correction, and then another for registration. For example, it is not uncommon for PET preprocessing that a single frame is chosen for motion correction, and then once the data is motion corrected, then the motion-corrected data is either averaged or summed across frames to create a new reference that is then used for co-registration. So we just need to be absolutely clear on the meaning of `_petref`.

**Comment by effigies** [comment]
Definitely a good point that the motion correction and coregistration references need not be the same. In BOLD, we generally want to do motion correction with an uncorrected reference, but then use a distortion-corrected reference for coregistration. Assuming we're okay with a common suffix for these two cases, it's just a matter of a `desc-` field.
